### ISSUE TITLE
[Bug] Navlayers: Technique and Layer classes improperly set metadata property

### ISSUE THREAD

**Issue by pljoel** [mitre-attack/mitreattack-python#63]
## Expected Behavior
Navlayers technique and layer classes should not error out when setting their metadata property.

## Actual Behavior
Navlayers technique and layer classes fail when setting the metadata property.
Their metadata setter check for Metadata and MetaDiv are incorrect.

See Metadata and MetaDiv classes: [Link](https://github.com/mitre-attack/mitreattack-python/blob/master/mitreattack/navlayers/core/metadata.py)
See Navlayers Technique metadata setter: [Link](https://github.com/mitre-attack/mitreattack-python/blob/master/mitreattack/navlayers/core/technique.py#L107-L110)
See Navlayers Layer metadata setter: [Link](https://github.com/mitre-attack/mitreattack-python/blob/master/mitreattack/navlayers/core/layerobj.py#L309-L313)

## Steps to Reproduce the Problem
1. Python 3
```
from mitreattack import navlayers

layerA = navlayers.Layer(init_data={'name': "Layer A", 'domain':"enterprise-attack"})
layer_technique = navlayers.Technique(tID="T1003")
layer_technique.metadata = [navlayers.Metadata(name="Metadata", value="1"), navlayers.MetaDiv(active=True)]
```

Error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/me/.local/share/virtualenvs/test-XrAZKEUD/lib/python3.8/site-packages/mitreattack/navlayers/core/technique.py", line 107, in metadata
    if "divider" in entry:
TypeError: argument of type 'Metadata' is not iterable
```

## Possible Solution

1. Check if metadata entries in the list are either Metadata or MetaDiv objects.
Change [technique.py line 107 to 110](https://github.com/mitre-attack/mitreattack-python/blob/master/mitreattack/navlayers/core/technique.py#L107-L110):
```
try:
    for entry in metadata:
        if isinstance(entry, Metadata) or isinstance(entry, MetaDiv):
            self.__metadata.append(entry)
        else:
            pass  # Object in the list was not of Metadata or MetaDiv classes
```
2. Unsure why, but the current version of the code duplicates Metadata and MetaDiv objects, so if that's still the way to go:
```
try:
    for entry in metadata:
        if isinstance(entry, MetaDiv):
            self.__metadata.append(MetaDiv(entry.state))
        elif isinstance(entry, Metadata):
            self.__metadata.append(Metadata(entry.name, entry.value))
        else:
            pass  # Object in the list was not of Metadata or MetaDiv classes
```

**Comment by ArtificialErmine** [comment]
Thanks for noticing this and bringing it to our attention! You're right - the system doesn't handle actual Metadata/MetaDiv arguments as classes when adding information, when it definitely should.

**Comment by pljoel** [comment]
Fixed with #66 .
